### PR TITLE
Add DID Auth support

### PR DESCRIPTION
**Pull request by ilanolkies**
Adds basic support for DID Auth (not configurable urls)

### CODE REVIEW

**Comment by jessgusclark** [comment]
Is the permissioned version working? When I run it, I get the error below.

```
 [nodemon] starting `node sample/back/index.js --permissioned`
[2] INFO: Accepting connections at http://localhost:3006
[1] INFO: Accepting connections at http://localhost:3005
[0] node sample/launcher.js backend exited with code 0
[3] C:\Users\jessg\Documents\gitprojects\rif\rLogin\rLogin\sample\back\index.js:22
[3] if (process.argv.length > 2 && process.argv[2] === '--permissioned') challenge.sdr = ['EmailCredential']
[3]                                                                      ^
[3]
[3] ReferenceError: challenge is not defined
[3]     at Object.<anonymous> (C:\Users\jessg\Documents\gitprojects\rif\rLogin\rLogin\sample\back\index.js:22:70)
[3]     at Module._compile (internal/modules/cjs/loader.js:1256:30)
[3]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1277:10)
[3]     at Module.load (internal/modules/cjs/loader.js:1105:32)
[3]     at Function.Module._load (internal/modules/cjs/loader.js:967:14)
[3]     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
[3]     at internal/main/run_main_module.js:17:47
[3] [nodemon] app crashed - waiting for file changes before starting...
```

File: sample/back/index.js [hunk 1]
```diff
@@ -1,33 +1,25 @@
...
-
-app.post('/request_auth', function (req, res) {
-  console.log(req.body.did)
-  challenge.challenge = Math.floor(Math.random() * 9999999999999)
-  res.status(200).send(challenge)
-})
-
-app.post('/auth', function (req, res) {
-  console.log(req.body.response)
+const authMiddleware = didAuth.default({ serviceDid, serviceSigner, serviceUrl, challengeSecret })(app)
```
Comment (javiesses) [inline sample/back/index.js:18]: `.default` :(

**Review by javiesses — APPROVED** [review]
LGTM

File: sample/back/index.js (hunk 1 above)
Comment (ilanolkies) [reply sample/back/index.js:18]: 🤷‍♀️

**Pull request merged**